File: Deniz/day_3.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        result = [[]]

        for num in nums:
            currentSize = len(result)
            for i in range(currentSize):
                newSubset = result[i][:] + [num]
                result.append(newSubset)

        return result

File: Deniz/day_3.go
package main

import "fmt"

func subsets(nums []int) [][]int {
    result := [][]int{{}}
    
    for _, num := range nums {
        currentSize := len(result)
        for i := 0; i < currentSize; i++ {
            newSubset := make([]int, len(result[i]), len(result[i])+1)
            copy(newSubset, result[i])
            newSubset = append(newSubset, num)

            result = append(result, newSubset)
        }
    }
    
    return result
}